### ISSUE TITLE
PHP Warning:  stream_socket_client(): SSL operation failed with code 1

### ISSUE THREAD

**Issue by Hydranet** [spotweb/spotweb#613]
I have been getting the following message when retrieving new spots for my spotweb installation since last night:
```
PHP Warning:  stream_socket_client(): SSL operation failed with code 1. OpenSSL Error messages:
error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed in /data/www/spotweb/html/vendor/spotweb/nntp/src/Protocol/Client.php on line 729
PHP Warning:  stream_socket_client(): Failed to enable crypto in /data/www/spotweb/html/vendor/spotweb/nntp/src/Protocol/Client.php on line 729
PHP Warning:  stream_socket_client(): unable to connect to ssl://news.newsleecher.com:563 (Unknown error) in /data/www/spotweb/html/vendor/spotweb/nntp/src/Protocol/Client.php on line 729


```
When I run it manually I get this:
```
Removing Spot information which is beyond retention period,, done
Last retrieve at Wed 09 Sep 2020 10:00:35 AM CEST
Retrieving new Spots from server news.newsleecher.com...
PHP Warning:  stream_socket_client(): SSL operation failed with code 1. OpenSSL Error messages:
error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed in /data/www/spotweb/html/vendor/spotweb/nntp/src/Protocol/Client.php on line 729
PHP Warning:  stream_socket_client(): Failed to enable crypto in /data/www/spotweb/html/vendor/spotweb/nntp/src/Protocol/Client.php on line 729
PHP Warning:  stream_socket_client(): unable to connect to ssl://news.newsleecher.com:563 (Unknown error) in /data/www/spotweb/html/vendor/spotweb/nntp/src/Protocol/Client.php on line 729
SpotWeb v0.68.31.33 on PHP v7.2.24 crashed

Fatal error occured while connecting to the newsserver:
  (-1) Error while connecting to server (server did not respond) [response: ""] [response: ""]


#0 /data/www/spotweb/html/lib/services/Nntp/Services_Nntp_Engine.php(143): Services_Nntp_Engine->connect()
#1 /data/www/spotweb/html/lib/services/Retriever/Services_Retriever_Base.php(118): Services_Nntp_Engine->selectGroup('free.pt')
#2 /data/www/spotweb/html/lib/services/Retriever/Services_Retriever_Base.php(298): Services_Retriever_Base->connect(Array)
#3 /data/www/spotweb/html/retrieve.php(142): Services_Retriever_Base->perform()
#4 {main}

Finished retrieving spots.

```

I use the same usenet account for my usenet downloader, when I do the server test there it succeeds, and also I am able to download like normaly. So there doesn't seem to be a problem with my usenet provider.

I also tried disabling ssl and then I get the same results:
```
Removing Spot information which is beyond retention period,, done
Last retrieve at Wed 09 Sep 2020 10:30:53 AM CEST
Retrieving new Spots from server news.newsleecher.com...
SpotWeb v0.68.31.33 on PHP v7.2.24 crashed

Fatal error occured while connecting to the newsserver:
  (-1) Failed to read from socket...! [response: ""] [response: ""]


#0 /data/www/spotweb/html/lib/services/Nntp/Services_Nntp_Engine.php(143): Services_Nntp_Engine->connect()
#1 /data/www/spotweb/html/lib/services/Retriever/Services_Retriever_Base.php(118): Services_Nntp_Engine->selectGroup('free.pt')
#2 /data/www/spotweb/html/lib/services/Retriever/Services_Retriever_Base.php(298): Services_Retriever_Base->connect(Array)
#3 /data/www/spotweb/html/retrieve.php(142): Services_Retriever_Base->perform()
#4 {main}

Finished retrieving spots.
```

 I run the development branch but in the logs of the recent changes I don't see anything that could point out to one of the changes being the problem.

**Comment by Sweepr** [comment]
If you run locally on your computer then the most common reason for messed up TLS certificates are antivirus applications intercepting nntp traffic, check if there are no firewall, antivirus or plugins messing with your traffic.

**Comment by Sweepr** [comment]
If you run without encryption (port 119) does that run normally?

**Comment by Hydranet** [comment]
> If you run locally on your computer then the most common reason for messed up TLS certificates are antivirus applications intercepting nntp traffic, check if there are no firewall, antivirus or plugins messing with your traffic.

I don't run it locally on my pc, but I run it on a VPS, which is hosted elsewhere. I will try that.

**Comment by Sweepr** [comment]
@hydranet Ok, there is still a chance for somereason port 563 is blocked on something else on the VPS changed overnight during some update.

**Comment by Sweepr** [comment]
Also running SSL using port 563 on my dev branch, not seeing any issue.

**Comment by Hydranet** [comment]
I switched it to port 119 without encryption and then it runs fine, then I switched back to port 563 but with TLS instead of SSL that ran fine also. I  finally then switched it back to 563 with SSL encryption and now it runs without crashing again. So I am not exactly sure what the problem was because I didn't change anything. Only thing I can think of that maybe my usenet provider had a problem with their ssl certificate, but there is no  way to test that now since it's working again. I should have tried unchecking "Verify name (CN) on certificate" in the first place to see if was my usenet provider ssl certificat was the problem. I also would have expected my usenet downloader to have tripped over that, so I have no idea what the problem was. Strangest thing ever.
